include/grpc++/auth_context.h

```diff
@@ -34,15 +34,42 @@
 #ifndef GRPCXX_AUTH_CONTEXT_H
 #define GRPCXX_AUTH_CONTEXT_H
 
+#include <iterator>
 #include <vector>
 
 #include <grpc++/config.h>
 
+struct grpc_auth_context;
```

Comment:
I think that having another AuthPropertyIterator public header makes sense. This way the API remains kind of clean.